### PR TITLE
fix: ensure connection auth details are sent to w2d correctly (fix #22)

### LINKED ISSUE #22: Latest Dev Build of Master can't connect to Sauce Labs and BrowserStack
## The problem

I'm using the latest master and execute the following steps:

1. `npm i` in the root
1. in a new terminal window `cd inspector && npm i && npm run dev`
1. set up a connection with Sauce Labs by providing my username and accesskey + create a config that works with AD 1.18.3

When I try to connect to Sauce Labs I get this error
image

and the logs
image

## Environment

- I am running Appium Desktop version _latest master_.
- I am on (pick one):
    - [x] Mac
    - [ ] Windows
    - [ ] Linux

### PR DESCRIPTION
**Pull request by jlipps**

### CODE REVIEW

File: app/renderer/actions/Session.js [hunk 1]
```diff
@@ -410,15 +410,19 @@ export function newSession (caps, attachSessId = null) {
...
     const serverOpts = {
-      hostname,
+      hostname: host,
       port: parseInt(port, 10),
       protocol: https ? 'https' : 'http',
       path,
       connectionRetryCount: CONN_RETRIES,
     };
 
+    if (username && accessKey) {
```
Comment (KazuCocoa) [inline app/renderer/actions/Session.js:421]: 👍 
https://webdriver.io/docs/options/#user

**Review by KazuCocoa — APPROVED** [review]

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by wswebcreation** [comment]
Hi @jlipps 

I've downloaded master and added your fix in it and it doesn't seem to work

image

**Comment by jlipps** [comment]
That's a strange error. Well if you want to DM me some access information and caps, I can try to troubleshoot on my end.

**Comment by wswebcreation** [comment]
Hi @jlipps 

I'll mail some details, but you can just use random strings. Then the error will also happen

**Comment by KazuCocoa** [comment]
probably https://stackoverflow.com/questions/39490904/errorfailed-to-execute-atob-on-window-the-string-to-be-decoded-is-not-corr/45192776 ...?

**Comment by KazuCocoa** [comment]
I guess parsing desiredCapabilities with `:` may have an issue...?
In sauce's tab, when I added `{a: 'b'}` capability, it returned invalid caps error.
But when I added `'a:a': 'b'` capability, the above error returned. Let me check further.

**Comment by shibupanda** [comment]
When I removed below it started working for me.
`    if (username && accessKey) {
      serverOpts.user = username;
      serverOpts.key = accessKey;
    }`

**Comment by KazuCocoa** [comment]
I guess https://github.com/appium/appium-inspector/pull/118#issuecomment-914027306 means https://github.com/appium/appium-inspector/pull/129. Could you try out the branch? @shibupanda

**Comment by shibupanda** [comment]
> I guess [#118 (comment)](https://github.com/appium/appium-inspector/pull/118#issuecomment-914027306) means #129. Could you try out the branch? @shibupanda

I checked and it worked. Thanks @KazuCocoa

File: app/renderer/actions/Session.js (hunk 1 above)
Comment (KazuCocoa) [reply app/renderer/actions/Session.js:433]: mm, these `user` and `key` is an issue...?
When I comment out here and sent a request with SauceLabs' tab, a different error returned.
So I guess this option did not work...(?)

**Comment by jlipps** [comment]
the remaining fix for Sauce is: https://github.com/webdriverio/webdriverio/pull/7401, it will need to be merged, published, merged into w2d, published, then merged into the inspector as well

**Comment by jlipps** [comment]
@testingbot do you mind signing our new CLA, please?

**Comment by KazuCocoa** [comment]
Thank you! @testingbot 
This PR is ready for merge cc @jlipps

**Pull request merged**